consensus/src/block_storage/sync_manager.rs

```diff
@@ -64,12 +64,24 @@ impl BlockStore {
     /// Check if we're far away from this ledger info and need to sync.
     /// This ensures that the block referred by the ledger info is not in buffer manager.
     pub fn need_sync_for_ledger_info(&self, li: &LedgerInfoWithSignatures) -> bool {
+        let block_not_exist = self.ordered_root().round() < li.commit_info().round()
+            && !self.block_exists(li.commit_info().id());
         // TODO move min gap to fallback (30) to config, and if configurable make sure the value is
         // larger than buffer manager MAX_BACKLOG (20)
-        (self.ordered_root().round() < li.commit_info().round()
-            && !self.block_exists(li.commit_info().id()))
-            || self.commit_root().round() + 30.max(2 * self.vote_back_pressure_limit)
-                < li.commit_info().round()
+        let max_commit_gap = 30.max(2 * self.vote_back_pressure_limit);
+        let minimum_commit_round = li.commit_info().round().saturating_sub(max_commit_gap);
+
+        if let Some(pre_commit_status) = self.pre_commit_status() {
+            let mut status_guard = pre_commit_status.lock();
+            if block_not_exist || status_guard.round() < minimum_commit_round {
+                status_guard.pause();
```

Comment:
Elaborate why pre_commit won't go beyond target and clarify that pre_commit is still going for in flight LIs.